core/src/main/java/org/testcontainers/images/RemoteDockerImage.java

```diff
@@ -95,4 +95,10 @@ protected final String resolve() {
             throw new ContainerFetchException("Failed to get Docker client for " + imageName, e);
         }
     }
+
+    @ToString.Include(name = "imageName", rank = 1)
```

Comment:
please add more tests for the scenario where `imageNameFuture` fails

Reply:
Still need to do this...

Reply:
Took a crack at it with https://github.com/testcontainers/testcontainers-java/pull/2450/commits/9b316322b2f323b32dbc861253154389485eb094

Reply:
btw why did you remove Lombok's toString? I really liked the `@ToString.Include` approach :)

Reply:
Seemed like I needed a place to add logic about using imageName if available, and falling back to imageNameFuture.

Reply:
You can create a special method for it (marked with `@ToString.Include`), without having to "delombok" the whole `toString`

Reply:
Yeah, it seems obvious now.  Lemme get the isDone part and get back to this.

Reply:
I'm staring at it again now and I see a potential struggle.  If I use `@ToString.Include` the "key" in the resulting string is always the same.  I no longer get to choose whether to have `imageName=foo` when I have an image name, and fall back to `imageNameFuture=...`.

But maybe it's OK, or even better, to have `imageName=` in both cases?

Reply:
@dbyron0 could you please share examples of the `toString` call of both options?

Reply:
With `@ToString.Include`:
* when imageName is available: `RemoteDockerImage{imageName=ibemazgg:latest, imagePullPolicy=DefaultPullPolicy(), dockerClient=LazyDockerClient.INSTANCE}`
* when imageName isn't available: `RemoteDockerImage{imageName=org.testcontainers.images.RemoteDockerImage$1@5d01486, imagePullPolicy=DefaultPullPolicy(), dockerClient=LazyDockerClient.INSTANCE}`

Without lombok:
* when imageName is available: `RemoteDockerImage{imageName=ibemazgg:latest, imagePullPolicy=DefaultPullPolicy(), dockerClient=LazyDockerClient.INSTANCE}`
* when imageName isn't available: `RemoteDockerImage{imageNameFuture=org.testcontainers.images.RemoteDockerImage$1@5d01486, imagePullPolicy=DefaultPullPolicy(), dockerClient=LazyDockerClient.INSTANCE}`

Reply:
I kinda like the Lombok version :)

side note: I would also exclude `dockerClient`.

Reply:
Fair enough: https://github.com/testcontainers/testcontainers-java/pull/2450/commits/1a31c5ce6739f66d38918253ef0c173a09f9e4d9

Would love a hand figure out the test failure / calling isDone.